core/services/relay/evm/event_binding.go

```diff
@@ -0,0 +1,272 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/codec"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+type eventBinding struct {
+	address        common.Address
+	contractName   string
+	eventName      string
+	lp             logpoller.LogPoller
+	hash           common.Hash
+	codec          commontypes.RemoteCodec
+	pending        bool
```

Comment:
I still think we should probably change `pending` to `finalized` (and invert the logic) to avoid confusion with the pending transaction pool.  This is a flag indicating whether to return finalized blocks or all blocks.  I also think it would be better if the default were to return all (mined) blocks.  (Under no circumstances would we ever return a pending block in the sense of pending transactions that haven't been mined yet.)

Reply:
Are you okay with that @jmank88 ?

Reply:
I don't have any opinions on these internals, but where does this connect to the UX? That is where the default value is important.

Reply:
Leaving for now, merged `common` already.  Can be updated later if needed.